onebusaway-android/src/google/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -346,6 +349,23 @@ protected void showDialog(int id) {
         MapDialogFragment.newInstance(id, this).show(getFragmentManager(), MapDialogFragment.TAG);
     }
 
+    private void setLocationByZip(final String zip) {
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                Location l = LocationUtils.getLocationFromZip(getContext(), zip);
+                if (l != null) {
+                    mLocationHelper.onLocationChanged(l);
+                    setMyLocation(l, true, true);
+                } else {
+                    Toast.makeText(getContext(), getString(R.string.main_ziplocation_error),
```

Comment:
`getContext()` should be `getApplicationContext()`, as the Toast could trigger when this Activity is destroyed.